docs/current_state_specification.md

```diff
@@ -0,0 +1,153 @@
+### Current State Representation
+
+The cloud backend (or any OTA update system) needs to keep track of the current state of the device. This current state information is relevant as input for the context pipeline for compiling future desired states and acts as an indicator of problems in applying a given desired state, if the current state differs from the desired state, which may not be reached.
+
+For representing the current state of the device, the approach of a holistic inventory of hardware and software nodes will be applied. This model is using a graph structure consisting of three element types - hardware nodes, software nodes, associations.
+
+### Current State Data Model
+
+The following table describes all supported properties and sections of the Current State specification:
+
+| Property | Type | Description |
+| - | - | - |
+| **General properties** | | |
+| hardwareNodes | JSON array | Inventory for a list of hardware nodes |
+| softwareNodes | JSON array | Inventory for a list of software nodes |
+| associations | JSON array | List of mappings between the inventory nodes |

```

Comment:
What is the semantic of such an association? Is it something like "runs on" or "managed by"? Does the semantic change depending on the involved nodes, e.g., the meaning is different for an association between a hardware node and software node compared to the association between two software nodes?

Reply:
Hi @eriksven , the general inventory model does not deliberately define any semantics of an association, the model defines just a link between two nodes (either hardware nodes, or software nodes). 
Any semantics are to be be defined on solution level. 
For the update manager case and OTA updates, we have agreed to have the UpdateManager (software node, type APPLICATION) as root, it has associations to all the update agents that shall be represented as software nodes (type APPLICATION) too. The associations of the update agent software nodes are specific for the given domain, e.g. containers update agents has all the available containers in the system as software nodes (type CONTAINER) and associations to them; [self-update agent](https://github.com/eclipse-leda/leda-contrib-self-update-agent] has OS image as software node with type IMAGE, etc...
E.g.

Reply:
Thanks for the explanation, maybe it makes sense to add this somewhere in the document, especially the part that an association "only" defines a link and not a fixed semantic.

Reply:
Some extra text added in the table description. Added a link to the semantics followed for the UpdateManager solution.